# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/wifi/nsWifiScannerDBus.h
# Commit: 35c408b9531d
# Full Hash: 35c408b9531db01c71dee288cab0a02a633fb535
# Author: David Parks <daparks@mozilla.com>
# Date: 2023-05-13 21:42:25
# Regressor Bug: 1810421
# File Overlap Count: 2
# Description:
#   Bug 1810421: Refactor wifi scanning to consolidate generic code r=necko-reviewers,kershaw,valentin
#   
#   Reorganizes the wifi-scanning code and makes more of it platform-generic
#   to ease the transition from polling the wifi to usually scanning only on
#   network changes.  This is essentially just moving files/code around and
# ==============================================================================

diff -r 029be11fa3d8 -r 35c408b9531d netwerk/wifi/nsWifiScannerDBus.h
--- a/netwerk/wifi/nsWifiScannerDBus.h	Sat May 13 00:11:51 2023 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,44 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef NSWIFIAPSCANNERDBUS_H_
-#define NSWIFIAPSCANNERDBUS_H_
-
-#include "nsCOMArray.h"
-
-#define DBUS_API_SUBJECT_TO_CHANGE
-#include <dbus/dbus.h>
-#include <dbus/dbus-glib-lowlevel.h>
-
-class nsWifiAccessPoint;
-
-namespace mozilla {
-
-class nsWifiScannerDBus final {
- public:
-  explicit nsWifiScannerDBus(nsCOMArray<nsWifiAccessPoint>* aAccessPoints);
-  ~nsWifiScannerDBus();
-
-  nsresult Scan();
-
- private:
-  nsresult SendGetDevices();
-  nsresult SendGetDeviceType(const char* aPath);
-  nsresult SendGetAccessPoints(const char* aPath);
-  nsresult SendGetAPProperties(const char* aPath);
-  nsresult IdentifyDevices(DBusMessage* aMsg);
-  nsresult IdentifyDeviceType(DBusMessage* aMsg, const char* aDevicePath);
-  nsresult IdentifyAccessPoints(DBusMessage* aMsg);
-  nsresult IdentifyAPProperties(DBusMessage* aMsg);
-  nsresult StoreSsid(DBusMessageIter* aVariant, nsWifiAccessPoint* aAp);
-  nsresult SetMac(DBusMessageIter* aVariant, nsWifiAccessPoint* aAp);
-  nsresult GetDBusIterator(DBusMessage* aMsg, DBusMessageIter* aIterArray);
-
-  RefPtr<DBusConnection> mConnection;
-  nsCOMArray<nsWifiAccessPoint>* mAccessPoints;
-};
-
-}  // namespace mozilla
-
-#endif  // NSWIFIAPSCANNERDBUS_H_